app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -19,41 +20,40 @@ class DrawingViewModel : ViewModel() {
         val editIndex: Int?
     )
 
-    private var isColorPickerSheetOpen = false
+    val layerNames: MutableList<String> = mutableListOf()
 
+    private var isColorPickerSheetOpen = false
     private var chosenColor: Int? = null
-
-    var activeBitmap: Bitmap? = null
-
+    private var isLayerDialogOpen = false   //new dialog
+    private var isDeleteDialogOpen = false //new delete dialog
+    private var isViewingLayerSheetOpen = false       //edit layer
     private var hsv = Hsv(0f, 0f, 0f)
 
+    var activeBitmap: Bitmap? = null
     var circleColor: Int = Color.BLACK
 
     //used to help determine rgb values for consistency
     var activeColor: Int = Color.BLACK
-    val layerNames: MutableList<String> = mutableListOf()
+    var currentLayerDialogText: String? = null
+
+    //view inside edit layer
+    var layerViewEditDialogIndex: Int? = null
```

Comment:
Private. Don't leak information outside that doesn't need to be there

Reply:
made both private, also introduced getters and setters for both.